### ISSUE TITLE
Ignorar pastas na geração do HTML

### ISSUE THREAD

**Issue by douglastelles** [AlencarGabriel/ProtheusDoc-VsCode#42]
Olá, primeiramente parabéns pela extensão. Tem sido muito útil para documentação dos meus projetos demonstrando um algo nível de profissionalidade ao apresentar o HTML com a documentação dos programas.

Utilizo muito a opção de gerar gerar a documentação por pastas, assim consigo englobar todo o projeto. Tentei encontrar nas configurações ou nas issues como ignorar uma pasta especifica semelhante ao .gitignore, porém não consegui identificar. Existe essa funcionalidade implementada?

Caso não exista, seria possível a implementação? Não conheço muito TypeScript mas se for o caso posso tentar submeter uma alteração para análise, cheguei a imaginar uma possível validação do repositório no arquivo extension.ts dentro de addGenerateHTMLFolder.

Para melhorar o entendimento, estou anexando uma imagem de um repositório de um cliente meu, onde tenho dentro do diretório /temp/ os arquivos de consulta ou testes que não não interessantes manter no repositório.

Grato pela extensão.

![exemplo-diretorio-exclude](https://user-images.githubusercontent.com/58710528/98126788-bb38ff80-1e94-11eb-906d-ea1cd8f58c4e.png)

**Comment by AlencarGabriel** [comment]
Olá @douglastelles ,

desde já muito obrigado pelo feedback e pela excelente sugestão de melhoria.

Realmente, hoje não há nada que permita realizar um ignore de algumas pastas na geração da documentação HTML.

Penso que poderia ser implementado uma configuração utilizando o conceito de _Glob patterns_, padrão do VsCode, e utilizado por muitas extensões. Este conceito permitiria implementar uma série de padrões (por arquivo, extensão, pasta, etc...) e funciona como um *.ignore, mas nas configurações do VsCode.

Mas nesses últimos meses tenho estado bastante atarefado, e não consegui fazer mais nenhuma melhoria nas extensões que contribuo. Caso deseje construir a melhoria seria um prazer ajudá-lo e avalia-la. Se estiver tudo certo com certeza farei a atualização da versão com esta.

**Comment by douglastelles** [comment]
> Olá @douglastelles ,
> 
> desde já muito obrigado pelo feedback e pela excelente sugestão de melhoria.
> 
> Realmente, hoje não há nada que permita realizar um ignore de algumas pastas na geração da documentação HTML.
> 
> Penso que poderia ser implementado uma configuração utilizando o conceito de _Glob patterns_, padrão do VsCode, e utilizado por muitas extensões. Este conceito permitiria implementar uma série de padrões (por arquivo, extensão, pasta, etc...) e funciona como um *.ignore, mas nas configurações do VsCode.
> 
> Mas nesses últimos meses tenho estado bastante atarefado, e não consegui fazer mais nenhuma melhoria nas extensões que contribuo. Caso deseje construir a melhoria seria um prazer ajudá-lo e avalia-la. Se estiver tudo certo com certeza farei a atualização da versão com esta.

Certo, vou dar uma estudada sobre o que sugeriu e qualquer duvido tento entrar em contato para seguirmos com a melhoria.
